### PR TITLE
changeable epsilon for (typeof Quaternion).FromUnitVectorsToRef

### PR DESCRIPTION
**Pull request by nekochanoide**
for my use case epsilon is to big, so right now I have to copy entire method and use 1e-7 epsilon which works fine for me
I hope optional parameter for epsilon is fine

### CODE REVIEW

**Review by RaananW — APPROVED** [review]

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13983/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13983/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13983/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13983/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13983/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13983/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13983/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13983/merge#BCU1XR#0

File: packages/dev/core/src/Maths/math.vector.ts [hunk 1]
```diff
@@ -4664,10 +4664,10 @@ export class Quaternion {
      * @param result the quaternion to store the result
      * @returns the updated quaternion
      */
-    public static FromUnitVectorsToRef<T extends Quaternion>(vecFrom: DeepImmutable<Vector3>, vecTo: DeepImmutable<Vector3>, result: T): T {
+    public static FromUnitVectorsToRef<T extends Quaternion>(vecFrom: DeepImmutable<Vector3>, vecTo: DeepImmutable<Vector3>, result: T, epsilon = Epsilon): T {
```
Comment (Popov72) [inline packages/dev/core/src/Maths/math.vector.ts:4667]: The doc for `epsilon` is missing in the function description.

**Comment by nekochanoide** [comment]
I wonder what purpose this epsilon check serves? Is it optimization or function might break, if dot product is near zero.

**Comment by carolhmj** [comment]
> I wonder what purpose this epsilon check serves? Is it optimization or function might break, if dot product is near zero.

It's to avoid breaking because of limited precision: https://blog.demofox.org/2017/11/21/floating-point-precision/

**Pull request merged**